netbench-orchestrator/src/orchestrator/cli.rs

```diff
@@ -0,0 +1,121 @@
+
+use crate::orchestrator::{OrchError, OrchResult};
+use serde::Deserialize;
+use std::{
+    fs::File,
+    path::{Path, PathBuf},
+};
+
+#[derive(Clone, Debug)]
+pub struct OrchestratorConfig {
+    // netbench
+    netbench_scenario_filename: String,
+    netbench_scenario_filepath: PathBuf,
+
+    // cdk
+    pub cdk_config: CdkConfig,
+}
+
+// Used for parsing the scenario file generated by the s2n-netbench project
+#[derive(Clone, Debug, Default, Deserialize)]
+pub struct CdkConfig {
+    #[serde(rename(deserialize = "NetbenchInfraPrimaryProd"))]
+    resources: CdkResources,
+}
+
+#[derive(Clone, Debug, Default, Deserialize)]
+#[serde(rename_all(deserialize = "camelCase"))]
+struct CdkResources {
+    output_netbench_runner_log_group: String,
+    output_netbench_runner_public_logs_bucket: String,
+    output_netbench_runner_private_src_bucket: String,
+    output_netbench_cloudfront_distribution: String,
+    output_netbench_runner_instance_profile: String,
+    output_netbench_subnet_tag_key: String,
+    output_netbench_subnet_tag_value: String,
+    output_netbench_infra_primary_prod_region: String,
+}
+
+impl OrchestratorConfig {
+    pub fn netbench_scenario_filename(&self) -> &str {
+        &self.netbench_scenario_filename
+    }
+
+    pub fn netbench_scenario_file_stem(&self) -> &str {
+        self.netbench_scenario_filepath
+            .as_path()
+            .file_stem()
+            .expect("expect scenario file")
+            .to_str()
+            .unwrap()
+    }
+
+    pub fn cf_url(&self, unique_id: &str) -> String {
+        format!(
+            "{}/{}",
+            self.cdk_config.netbench_cloudfront_distribution(),
+            unique_id
+        )
+    }
+
+    pub fn s3_path(&self, unique_id: &str) -> String {
+        format!(
+            "s3://{}/{}",
+            self.cdk_config.netbench_runner_public_s3_bucket(),
+            unique_id
+        )
+    }
+
+    pub fn s3_private_path(&self, unique_id: &str) -> String {
+        format!(
+            "s3://{}/{}",
+            self.cdk_config.netbench_runner_private_s3_bucket(),
+            unique_id
+        )
+    }
+}
+
+impl CdkConfig {
+    pub fn netbench_runner_public_s3_bucket(&self) -> &String {
+        &self.resources.output_netbench_runner_public_logs_bucket
+    }
+
+    pub fn netbench_runner_private_s3_bucket(&self) -> &String {
+        &self.resources.output_netbench_runner_private_src_bucket
+    }
+
+    pub fn netbench_cloudfront_distribution(&self) -> &String {
+        &self.resources.output_netbench_cloudfront_distribution
+    }
+
+    pub fn netbench_runner_log_group(&self) -> &String {
+        &self.resources.output_netbench_runner_log_group
+    }
+
+    pub fn netbench_runner_instance_profile(&self) -> &String {
+        &self.resources.output_netbench_runner_instance_profile
+    }
+
+    pub fn netbench_runner_subnet_tag_key(&self) -> String {
+        format!("tag:{}", self.resources.output_netbench_subnet_tag_key)
```

Comment:
Why is `tag:{}` necessary? Is this something that get's passed directly into an EC2 API? If so it might be nice to link some documentation about what it's doing.